src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,79 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that has a deadline.
+ */
+public class Deadline extends Task {
+
+    public static final String TAG = "D";
+    protected LocalDateTime dateAndTime;
+
+    /**
+     * Constructor for a Deadline.
+     * @param name Name of the task
+     * @param dateAndTime Deadline of the task
+     */
+    public Deadline(String name, String dateAndTime) {
+        super(name);
+
+        String[] dateAndTimeSplit = dateAndTime.split(" ");
```

Comment:
Perhaps using DateTimeFormatterBuilder to build a DateTimeFormatter that accepts custom formats, then using LocalDate.parse() with the formatter could be more effective